HotReload: Subscription feature preview (#4116)
dapr/docs: daprdocs/content/en/developing-applications/building-blocks/pubsub/howto-publish-subscribe.md

Place `subscription.yaml` in the same directory as your `pubsub.yaml` component. When Dapr starts up, it loads subscriptions along with the components.

{{% alert title="Note" color="primary" %}}

> Dapr can be made to "hot reload" declarative subscriptions whereby updates are picked up automatically without needing a restart.

Reviewer: ```suggestion
Dapr can be made to "hot reload" declarative subscriptions, whereby updates are picked up automatically without needing a restart.
```

This is enabled by via the [`HotReload` feature gate]({{< ref "support-preview-features.md" >}}).

> In-flight messages between Dapr and your application will be unaffected during hot reload events to prevent re-processing or loss of unprocessed messages.

Reviewer: ```suggestion
To prevent reprocessing or loss of unprocessed messages, in-flight messages between Dapr and your application are unaffected during hot reload events.
```

> This feature is currently in preview.

Reviewer: nit: I'd move this to the top of the note to make it a bit more prominent
